odoo/tests/runner.py

```diff
@@ -15,12 +24,42 @@ class OdooTestResult(unittest.result.TestResult):
     have based on the test object that is running.
     """
 
-    def __init__(self):
+    def __init__(self, suite_tests_count=0):
         super().__init__()
         self.time_start = None
         self.queries_start = None
         self._soft_fail = False
         self.had_failure = False
+        self.suite_tests_count = suite_tests_count
+
+        self.progressbar = progressbar and progressbar.ProgressBar(
+            widgets=[
+                progressbar.SimpleProgress(),
+                progressbar.MultiRangeBar("progress_counter", markers=[
+                    '\x1b[32m▣\x1b[39m',  # Passing
+                    '\x1b[31m▢\x1b[39m',  # Errors
+                    '\x1b[33m▢\x1b[39m',  # Failures
+                    '\x1b[37m▫\x1b[39m',  # Skipped
+                    ' '  # Not started
+                ]),
+            ],
+            max_value=suite_tests_count,
+            redirect_stdout=True,
+        ).start()
+
+    def __del__(self):
+        if self.progressbar:
+            self.progressbar.finish()
```

Comment:
I also noticed that the progressbar is sometimes copied right after a traceback is logged (I have been using this a few days in a big refactor branch)

Reply:
Another issue is that the logging is stopped during tests and is only flushed at the end.
Particularly annoying with tour tests where nothing is happening for a while in the console, and it is counter productive to the purpose which is to have a better idea of what is happening with a glance

Reply:
I had a similar issue especially when an install runs before a test where nothing is flushed before the starts of the test
If no test is run, the console just stays empty in my case. And it is even more broken if no test is ran

![image](https://user-images.githubusercontent.com/35262360/174241115-cc0cbb8a-d7b4-4440-9f2a-e2b864ec1813.png)

Reply:
Another issue: impossible to use `pdb` for instance. It is working, but not displaying anything (you see the commands and results after you exist the debugger and the tests finishes so that it can flush)

Reply:
Maybe it is better if we stick with the number display only, one can make a simple progress bar with an external program easily, like this one that I use for upgrade progress:
```bash
function o-upgrade() {
### module1[,mudule2,...] : upgrades selected modules and exits
  $ODOO_BIN `o-get-args` -u $1 --st --log-level info 2>&1 | awk '
    BEGIN {
      start=0; 
      trace=0; 
      len=80; # progressbar length
      total=1; # prevent division by zero
      for(i=1;i<=len;i++) # init progressbar
        printf "."; 
      printf "\r" 
    }
    /(CRITICAL|ERROR|WARNING)/ {
      trace=1; # error detected
      # erase progress bar
      printf "                                                                                          \n"; 
    } 
    /(INFO)/ {
      trace=0 # end of error
      # ok, redraw progressbar
      for(i=1;i<=len;i++) 
          printf (i/len <= loaded/total) ? ">" : "."; 
      printf int(loaded/total*100) "% " loaded "/" total "\r" ;
    }
    trace { # in error tracing
      print;
      next;
    } 
    /Loading module .*\([0-9]+/ {
      split($10, data, "[/)(]") 
      if (start == 0) # if this is the first report, set the starting point
        start=data[2];
      loaded = data[2]-start 
      total = data[3]-start
      for(i=1;i<=len;i++) # print progress bar
          printf (i/len <= loaded/total) ? ">" : "."; 
      printf int(loaded/total*100) "% " loaded "/" total "\r" ;
    } 
    END { 
      printf "\n";  
    } 
    '
  echo "Exited with error code: ${PIPESTATUS[0]}${pipestatus[1]}"
}
```